src/Service/Auth/auth.service.ts

```diff
@@ -0,0 +1,120 @@
+import { Request, Response, NextFunction } from "express";
+import { StatusCodes } from "http-status-codes";
+import { UserType } from "../../Models/user.model";
+import AuthRepository from "../../Repository/auth.repository";
+import validation from "../../Utils/ValidationUtil/validation"
+import {
+    BadRequestError, 
+    NotFoundError, 
+    UnAuthenticatedError
+} from "../../Utils/ErrorUtil"
+import bcrypt from "../../Utils/bcrypt.encrypt"
+
+
+// instances
+const authRepository = new AuthRepository;
+const validations = new validation;
+
+
+export default class AuthService {
+    public async registerUser(req: Request, res: Response, next: NextFunction) {
+        try {
+            const { userName, password, country, phoneNumber, email } = req.body;
+
+            const user = {
+                user_name: userName,
+                password_digest: password,
+                phone_number: phoneNumber,
+                email,
+                country: country,
+            };
+
+            const userEmail = await validations.where('email', user.email);
```

Comment:
I also see a sample code like this used in few other plaaces...What's the tradeoff with this approach?